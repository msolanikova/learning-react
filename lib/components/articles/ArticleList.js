import React, {useEffect} from 'react';
import Article from "./Article";

const ArticleList = ({articles, articleActions}) => {
    return articles ? (
        <div>
            {Object.values(articles).map(article =>
                <Article key={article.id} article={article} actions={articleActions} />
            )}
        </div>
    ) : (
        <div>No articles available</div>
    );
}

export default ArticleList;
