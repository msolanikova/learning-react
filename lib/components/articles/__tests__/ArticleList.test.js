import React from 'react';
import renderer from 'react-test-renderer';
import ArticleList from "../ArticleList";

describe('ArticleList', () => {
    const testProps = {
        articles: {
            a: {id: "a", authorId: "1", title: "A", body: "AA"},
            b: {id: "b", authorId: "2", title: "B", body: "BB"},
        },
        articleActions: {
            lookupAuthor: jest.fn().mockReturnValue({
                id: "1",
                firstName: "X",
                lastName: "Y",
                website: "https://test.com"
            })
        }
    }

    it('should render list correctly', () => {
        const tree = renderer.create(
            <ArticleList {...testProps} />
        ).toJSON();

        console.log(tree);

        expect(tree.children.length).toBe(2);
        expect(tree).toMatchSnapshot();
    })
});
