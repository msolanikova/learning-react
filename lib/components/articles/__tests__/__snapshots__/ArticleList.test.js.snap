// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList should render list correctly 1`] = `
<div>
  <div
    className="article"
  >
    <div>
      <b>
        A
      </b>
    </div>
    <div
      className="author"
    >
      by 
      <a
        href="https://test.com"
      >
        X
         
        Y
      </a>
    </div>
    <div
      className="body"
    >
      AA
    </div>
  </div>
  <div
    className="article"
  >
    <div>
      <b>
        B
      </b>
    </div>
    <div
      className="author"
    >
      by 
      <a
        href="https://test.com"
      >
        X
         
        Y
      </a>
    </div>
    <div
      className="body"
    >
      BB
    </div>
  </div>
</div>
`;
